File: core/tests/fixtures/mothers/book_mother.py
from src.domain.book import Book
from tests.fixtures.mothers.base import BaseMother
from .author_mother import AuthorMother
from .book_detail_mother import BookDetailMother
from .book_sale_mother import BookSaleMother
from .editorial_mother import EditorialMother


class BookMother(BaseMother):
    """
    A mother class for creating Book fixtures.
    """
    _ROMEO_AND_JULIET = {
        'title': 'Romeo and Juliet',
        'description':
            'Romeo and Juliet is a tragedy written by William Shakespeare early in his career about two young star-crossed lovers whose deaths ultimately reconcile their feuding families.',
        'author': AuthorMother.shakespeare,
        'editorial': EditorialMother.anaya,
        'detail': BookDetailMother.valid,
        'sale': BookSaleMother.valid,
        'created_by': 'test',
    }

    _DON_QUIJOTE = {
        'title': 'Don Quijote',
        'description':
            'Don Quijote de la Mancha is a Spanish novel by Miguel de Cervantes. It follows the adventures of a nobleman who reads so many chivalric romances that he loses his sanity and decides to become a knight-errant, reviving chivalry and serving his nation.',
        'author': AuthorMother.cervantes,
        'editorial': EditorialMother.anaya,
        'detail': BookDetailMother.quijote,
        'sale': BookSaleMother.valid,
        'created_by': 'test',
    }

    @classmethod
    def romeo_and_juliet(cls, **overrides) -> Book:
        """
        Creates a Book instance for "Romeo and Juliet" with optional overrides.
        """
        return cls._create(cls._ROMEO_AND_JULIET, overrides)

    @classmethod
    def don_quijote(cls, **overrides) -> Book:
        """
        Creates a Book instance for "Don Quijote" with optional overrides.
        """
        return cls._create(cls._DON_QUIJOTE, overrides)

    @classmethod
    def _create(cls, values: dict, overrides: dict) -> Book:
        """
        Creates a Book instance with the given values and overrides.
        """
        data = cls._merge(values, overrides)
        return Book.create(**data)

File: editor/tests/fixtures/mothers/book.mother.ts
import { BaseMother } from '@tests/fixtures/mothers/base/base-mother';
import { AuthorMother } from '@tests/fixtures/mothers/author.mother';
import { EditorialMother } from '@tests/fixtures/mothers/editorial.mother';
import { BookDetailMother } from '@tests/fixtures/mothers/book-detail.mother';
import { BookSaleMother } from '@tests/fixtures/mothers/book-sale.mother';
import { Book } from '@/books/model/book';
import { MotherMapping } from '@tests/fixtures/mothers/base/mother-mapping';
import { BookDescriptor } from '@/books/model/book-descriptor';

const ROMEO_AND_JULIET = {
  id: '0e5b4225-063e-4330-a84b-09d175ed0270',
  title: 'Romeo and Juliet',
  description:
    'Romeo and Juliet is a tragedy written by William Shakespeare early in his career about two young star-crossed lovers whose deaths ultimately reconcile their feuding families.',
  author: AuthorMother.shakespeare,
  editorial: EditorialMother.anaya,
  detail: BookDetailMother.valid,
  sale: BookSaleMother.valid,
  createdBy: 'test',
  createdAt: ['2026-04-28T10:25:36', MotherMapping.DATE],
};

const DON_QUIXOTE = {
  id: 'b177f648-d843-4ada-b7c2-f0155ebf87d7',
  title: 'Don Quijote',
  description:
    'Don Quijote de la Mancha is a Spanish novel by Miguel de Cervantes. It follows the adventures of a nobleman who reads so many chivalric romances that he loses his sanity and decides to become a knight-errant, reviving chivalry and serving his nation.',
  author: AuthorMother.cervantes,
  editorial: EditorialMother.anaya,
  detail: BookDetailMother.quijote,
  sale: BookSaleMother.valid,
  createdBy: 'test',
  createdAt: ['2026-04-28T10:25:36', MotherMapping.DATE],
};

type BookFixture = typeof ROMEO_AND_JULIET;

export class BookMother extends BaseMother {
  static romeoAndJuliet(this: void, overrides?: Partial<Book>): Book {
    return BookMother.create(ROMEO_AND_JULIET, overrides);
  }

  static donQuixote(this: void, overrides?: Partial<Book>): Book {
    return BookMother.create(DON_QUIXOTE, overrides);
  }

  protected static create(
    values: BookFixture,
    overrides?: Partial<Book>,
  ): Book {
    const fields: Book = BookMother.merge<BookFixture, Book>(values, overrides);
    return Book.from(fields);
  }
}

export class BookDescriptorMother {
  static romeoAndJuliet(this: void, overrides?: Partial<Book>): BookDescriptor {
    const book = BookMother.romeoAndJuliet(overrides);
    return new BookDescriptor(book.id, book.title, book.detail.isbn);
  }

  static donQuixote(this: void, overrides?: Partial<Book>): BookDescriptor {
    const book = BookMother.donQuixote(overrides);
    return new BookDescriptor(book.id, book.title, book.detail.isbn);
  }
}
